### PR TITLE
rpmb: return RPMB_RESULT_AUTH_KEY_NOT_PROGRAMMED if key not set 

### PR DESCRIPTION
**Pull request by vchong**

### CODE REVIEW

**Comment by vchong** [comment]
Depends on https://github.com/OP-TEE/optee_os/pull/3087.

Let me test this on QEMU first and post an update with the results.

**Comment by vchong** [comment]
For `rpmb emu: increase EMU_RPMB_SIZE_MULT to 2`, should we consider increasing this to 4 MB instead? This is the size of all the real rpmb devices (3) I've seen so far.

**Comment by vchong** [comment]
Link to `optee_os` details related to this PR.
https://github.com/OP-TEE/optee_os/pull/3087#issuecomment-504306575
https://github.com/OP-TEE/optee_os/pull/3087#discussion_r296324166

**Comment by vchong** [comment]
> The defines you're adding are never used except RPMB_RESULT_AUTH_KEY_NOT_PROGRAMMED so I don't really see the point.
I'd rather add only the one that is used in the commit that uses it.

Removed unused defines.

> Please remove the defines from the commit message.

Ok, will remove when adding tag.

**Comment by vchong** [comment]
All comments addressed

**Comment by vchong** [comment]
Added a patch to change AOSP RPMB device path.

File: tee-supplicant/src/rpmb.c [hunk 1]
```diff
@@ -341,6 +341,11 @@ static bool is_hmac_valid(struct rpmb_emu *mem, struct rpmb_data_frame *frm,
 	return true;
 }
 
+static uint16_t gen_msb1st_result(uint8_t byte)
+{
+	return (uint16_t)(byte << 8);
```
Comment (etienne-lms) [inline tee-supplicant/src/rpmb.c]: prefer cast before shifting:  `return (uint16_t)byte << 8;`

Comment (vchong) [reply tee-supplicant/src/rpmb.c]: Done. Thanks!

**Comment by vchong** [comment]
Added patch to debug print `dev_id`. Please continue review. Thanks!

**Comment by jforissier** [comment]
For "rpmb: add DMSG to print dev_id":

`Reviewed-by: Jerome Forissier <jerome.forissier@linaro.org>`

**Comment by jenswi-linaro** [comment]
Please squash all commits with their fixups.

**Comment by vchong** [comment]
@jenswi-linaro I'm still missing R-b tags for "rpmb: add missing defines for RPMB results", "rpmb emu: increase EMU_RPMB_SIZE_MULT to 2" and "rmpb: update AOSP RPMB device path".

**Comment by jforissier** [comment]
@vchong it doesn't matter ;-) I suppose Jens would like to see what we have here without having to figure out the fixups. In fact, for relatively small patches, I find it easier to review the whole series again in the latest form. For bigger patches however, it's good to work incrementally.

**Comment by vchong** [comment]
@jenswi-linaro @jforissier Rebased, squashed and added tags wherever appropriate. Please continue review. Thanks!

**Comment by jenswi-linaro** [comment]
Looks good. For the commits:
```
rpmb: add missing defines for RPMB results
rpmb emu: return RPMB_RESULT_AUTH_KEY_NOT_PROGRAMMED if key not set
rpmb emu: remove gratuitous line
rpmb: add DMSG to print dev_id
```
Please apply:
`Reviewed-by: Jens Wiklander <jens.wiklander@linaro.org>`

and for the commits:
```
rpmb emu: fix ioctl return type
rpmb emu: increase EMU_RPMB_SIZE_MULT to 2
rmpb: update AOSP RPMB device path
```
`Acked-by: Jens Wiklander <jens.wiklander@linaro.org>`

**Comment by vchong** [comment]
Rebased and tags applied. Ok to merge.

File: tee-supplicant/src/rpmb.c [hunk 2]
```diff
@@ -93,6 +93,7 @@ struct rpmb_data_frame {
 #define RPMB_RESULT_GENERAL_FAILURE		0x01
 #define RPMB_RESULT_AUTH_FAILURE		0x02
 #define RPMB_RESULT_ADDRESS_FAILURE		0x04
+#define RPMB_RESULT_AUTH_KEY_NOT_PROGRAMMED	0x07
```
Comment (etienne-lms) [inline tee-supplicant/src/rpmb.c]: dummy question: why not 0x3 or 0x5? below 0x7 are reserved?

Comment (jforissier) [reply tee-supplicant/src/rpmb.c]: 0x3, 0x5, 0x6 are defined by the eMMC spec but are not used in the code, that's why they do not need to be here.

Comment (etienne-lms) [reply tee-supplicant/src/rpmb.c]: ok, thanks. I should have re-read to comment few lines above.

**Pull request merged**